sdk/core/azure-core-serializer-json-jackson/src/main/java/com/azure/core/serializer/json/jackson/implementation/SemanticVersion.java

```diff
@@ -83,7 +83,7 @@ public static SemanticVersion parse(String version) {
 
             String prerelease = (patchEndIdx == extEndIdx) ? "" : version.substring(patchEndIdx + 1, extEndIdx);
             return new SemanticVersion(major, minor, patch, prerelease, version);
-        } catch (NumberFormatException | NullPointerException ignored) {
+        } catch (NumberFormatException ignored) {
```

Comment:
Are we not expecting an NPE here anymore? What are the cases where it could happen?

Reply:
The way the code was written this could never NPE